### ISSUE TITLE
cargo xtask install fails with database connection errors

### ISSUE THREAD

**Issue by mmoncure** [supabase-community/postgres_lsp#152]
Sorry, this likely an obvious problem, but I'm stuck!  

Getting (error: error communicating with database: Connection refused (os error 61))

## To Reproduce

```
error: error communicating with database: Connection refused (os error 61)
  --> crates/pg_schema_cache/src/versions.rs:17:9
   |
17 | /         sqlx::query_as!(
18 | |             Version,
19 | |             r#"select
20 | |   version(),
...  |
28 | |   current_setting('max_connections') :: int8 AS max_connections;"#
29 | |         )
   | |_________^
   |
   = note: this error originates in the macro `$crate::sqlx_macros::expand_query` which comes from the expansion of the macro `sqlx::query_as` (in Nightly builds, run with -Z macro-backtrace for more info)
```

## Expected behavior

I can connect with psql with no arguments due to how environment is set -- what is the build using to connect, and to what?

## System information

macOS

**Comment by mmoncure** [comment]
was able to get past this. problem is kinda/sorta on my end; I do a lot of docker development to different backends and local dev database was running on a custom port to avoid various conflicts.  resetting port back to standard port seemed to do the trick. 

Having said that, having direct build dependency on database connection minutia checked into the code base just seems a bit off.  At the vary least, it ought to pull in connection details from the environment if set IMO. Just my 0.2$

**Comment by psteinroe** [comment]
hey! thanks for pulling through. the codebase is currently in a very dynamic state, sorry about that. we will set it up and document it properly once the alpha is finally released.

**Comment by mmoncure** [comment]
Hey, it's all good, I've read all your documentation and I think I understand where you're at.   I was able to progress to vs.code integration, and get messages flowing back and forth, then punted.

I did have to hack the log location:
-    let path = PathBuf::from_str("pglsp.log").expect("Opened the log file.");
+    let path = PathBuf::from_str("/tmp/pglsp.log").expect("Opened the log file.");

in order to get the server to start as will most people since installed server binary would be in secure folder (/usr/bin, etc)

I think, in order to get better engagement, hence more support, I have some high level suggestions:

1. remove all postgres connection dependences from the build process, even if that means separating tests from the build

2. have server binary provide some logging to stderr, start up messages, and various issues, unless that is not allowed in lsp specification, maybe with debug flag for extra verbosity.   Be very verbose about connecting to database for server cache etc.

3. server binary should honor environment variables 

4. provide various tests that supply json rpc messages to the binary in sequence and verify responses, this is much more useful and accessible than having internal rust unit test.  This will also provide cleaner separation between your environment and external environments, which is where I think there may be a bit of trouble today, especially since at current state the editor integrations are unreliable.
